src/widgets/sparkline.rs

```diff
@@ -153,57 +175,127 @@ impl<'a> Widget for Sparkline<'a> {
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec![" ▁▂▃▄▅▆▇█xxx"]));
+    }
+
+    #[test]
+    fn it_renders_with_max_more_than_data_max() {
+        let widget = Sparkline::default()
+            .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            .max(16);
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec!["  ▁▁▂▂▃▃▄xxx"]));
+    }
+
+    #[test]
+    fn it_renders_with_max_less_than_data_max() {
+        let widget = Sparkline::default()
+            .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            .max(4);
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec![" ▂▄▆█████xxx"]));
+    }
+
+    #[test]
+    fn it_renders_with_multi_line() {
+        let widget = Sparkline::default().data(&[0, 1, 2, 3, 4, 5, 6, 7, 8]);
+        let buffer = render(widget, 15, 3);
+        assert_buffer_eq!(
+            buffer,
+            Buffer::with_lines(vec![
+                "      ▂▅█xxxxxx",
+                "   ▁▄▇███xxxxxx",
+                " ▃▆██████xxxxxx",
+            ])
+        );
+    }
+
+    #[test]
+    fn it_renders_with_multi_line_and_baseline() {
+        let widget = Sparkline::default()
+            .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            .show_baseline(true);
+        let buffer = render(widget, 15, 3);
+        assert_buffer_eq!(
+            buffer,
+            // this currently fails because the baseline logic doesn't clear
+            // the parts above the line
+            // "      ▂▅█xxxxxx",
+            // "   ▁▄▇███xxxxxx",
+            // " ▃▆██████▁▁▁▁▁▁",
+            Buffer::with_lines(vec![
+                "      ▂▅█      ",
+                "   ▁▄▇███      ",
+                " ▃▆██████▁▁▁▁▁▁",
+            ])
+        );
     }
 
     #[test]
     fn it_renders_left_to_right() {
         let widget = Sparkline::default()
             .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
             .direction(RenderDirection::LeftToRight);
-        let buffer = render(widget, 12);
-        assert_eq!(buffer, Buffer::with_lines(vec![" ▁▂▃▄▅▆▇█xxx"]));
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec![" ▁▂▃▄▅▆▇█xxx"]));
     }
 
     #[test]
     fn it_renders_right_to_left() {
         let widget = Sparkline::default()
             .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
             .direction(RenderDirection::RightToLeft);
-        let buffer = render(widget, 12);
-        assert_eq!(buffer, Buffer::with_lines(vec!["xxx█▇▆▅▄▃▂▁ "]));
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec!["xxx█▇▆▅▄▃▂▁ "]));
+    }
+
+    #[test]
+    fn it_renders_baseline() {
+        let widget = Sparkline::default()
+            .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            .show_baseline(true);
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec!["▁▁▂▃▄▅▆▇█▁▁▁"]));
```

Comment:
I'm unsure whether a sparkline with a baseline should render empty values as empty or with a baseline.
`"▁▁▂▃▄▅▆▇█▁▁▁"` Baseline even when empty or:
 `"▁▁▂▃▄▅▆▇█   "` No baseline when there is no value.
 
The second option makes more sense to me, or perhaps make it configurable (use an enum rather than bool for the config setting).